### PR TITLE
Convert-from-nuget should simplify framework restrictions if possible

### PR DESCRIPTION
**Pull request by forki**
implements #1107 

/cc @theimowski could you please take a short look? Thx

### CODE REVIEW

File: src/Paket.Core/NugetConvert.fs [hunk 1]
```diff
@@ -296,7 +296,9 @@ let createDependenciesFileR (rootDirectory : DirectoryInfo) nugetEnv mode =
 
             Paket.DependenciesFile(DependenciesFileParser.parseDependenciesFile dependenciesFileName newLines))
 
-    if File.Exists dependenciesFileName then read() else create()
+    match (if File.Exists dependenciesFileName then read() else create()) with
```
Comment (theimowski) [inline src/Paket.Core/NugetConvert.fs]: use `Trial.lif` (map) ?

File: src/Paket.Core/Simplifier.fs [hunk 2]
```diff
@@ -66,7 +66,7 @@ let simplify interactive environment = trial {
     let! lockFile = environment |> PaketEnv.ensureLockFileExists
 
     let flatLookup = lockFile.GetDependencyLookupTable()
-    let dependenciesFileRef = ref environment.DependenciesFile
+    let dependenciesFileRef = ref (environment.DependenciesFile.SimplifyFrameworkRestrictions())
```
Comment (theimowski) [inline src/Paket.Core/Simplifier.fs]: this will happen also on `paket simplify` command?

Comment (forki) [inline src/Paket.Core/Simplifier.fs]: on simplify and convert

File: src/Paket.Core/DependenciesFile.fs [hunk 3]
```diff
@@ -398,6 +398,27 @@ type DependenciesFile(fileName,groups:Map<GroupName,DependenciesGroup>, textRepr
 
     member __.Groups = groups
 
+    member this.SimplifyFrameworkRestrictions() = 
+        let transform (dependenciesFile:DependenciesFile) (group:DependenciesGroup) =
+            if group.Options.Settings.FrameworkRestrictions <> [] then dependenciesFile else
```
Comment (theimowski) [inline src/Paket.Core/DependenciesFile.fs]: what happens when there are global restrictions and standalone restrictions per package? The latter override global?

File: src/Paket.Core/DependenciesFile.fs [hunk 4]
```diff
@@ -398,6 +398,27 @@ type DependenciesFile(fileName,groups:Map<GroupName,DependenciesGroup>, textRepr
...
+            match group.Packages with
+            | [] -> dependenciesFile
+            | package::rest ->
+                let sameRequirements =
+                    rest |> Seq.forall (fun p' -> p'.Settings.FrameworkRestrictions = package.Settings.FrameworkRestrictions)
+
+                if not sameRequirements || package.Settings.FrameworkRestrictions = [] then dependenciesFile else
```
Comment (theimowski) [inline src/Paket.Core/DependenciesFile.fs]: must check if restrictions are empty? does it harm if you add empty restrictions globally?

File: src/Paket.Core/DependenciesFile.fs (hunk 3 above)
Comment (forki) [inline src/Paket.Core/DependenciesFile.fs]: we don't do anything in this case.
during normal run both restrictions work together as two filters.

File: src/Paket.Core/DependenciesFile.fs (hunk 4 above)
Comment (forki) [inline src/Paket.Core/DependenciesFile.fs]: yes we create a stupid line

**Comment by theimowski** [comment]
LGTM, wondering if `SimplifyFrameworkRestrictions` could be made a bit more idiomatic, but that's probably just some cosmetics

**Comment by forki** [comment]
What would you call it?

**Comment by theimowski** [comment]
don't mean name itself, but rather the logic - maybe replace ifs with pattern matching where possible - but as I said it's just cosmetics, not crucial

**Comment by theimowski** [comment]
:+1:

**Comment by forki** [comment]
we now allow to simplify common framework restrictions. see https://github.com/fsprojects/Paket/blob/master/tests/Paket.Tests/Simplifier/BasicScenarioSpecs.fs#L342

**Comment by theimowski** [comment]
neat!

**Pull request merged**